agent/consul/state/catalog.go

```diff
@@ -517,7 +517,11 @@ func (s *Store) deleteNodeTxn(tx *memdb.Txn, idx uint64, nodeName string) error
 	}
```

Comment:
Noting that this implementation will have confusing behavior during upgrades - if some servers have the new code and some don't existing requests may go from a higher index returned in X-CONSUL-INDEX (from old code path) to a lower (per service value) and back to higher, depending on whether the server upgraded became a leader or not. We will need to document this somewhere. I suggest the upgrade page https://www.consul.io/docs/upgrading.html where you can add a paragraph describing the old vs new behavior.

Reply:
DONE